README.md

```diff
@@ -48,298 +48,400 @@ The FWS Payments contract enables ERC20 token payment flows through "rails" - au
+-   **Stops Future Lockups:** The client's account `lockupRate` is immediately reduced by the rail's `paymentRate`. This is a crucial step that stops the client from accruing any *new* lockup obligations for this rail.
+-   **Frees Operator Allowances:** The operator's rate usage is decreased, freeing up their `rateAllowance` for other rails.
+
+**Validator Callback**
+
+If the rail has a validator, `terminateRail` makes a synchronous call to the `validator.railTerminated` function. This is a powerful mechanism:
+
+-   **Veto Power:** The validator can block the termination attempt entirely by reverting inside this callback. This gives the validator the ultimate say on whether a rail can be terminated, irrespective of who initiated the call.
+-   **Notification:** It serves as a direct notification to the validator that a rail it oversees is being terminated, allowing it to update its own internal state if needed.
+
+-   **Parameters**:
+    -   `railId`: The rail's unique identifier.
+-   **Requirements**:
+    -   Caller must be the rail's client (and have a fully funded account) or the rail's operator.
+    -   The rail must not have been already terminated.
 
 #### `modifyRailLockup(uint256 railId, uint256 period, uint256 lockupFixed)`
 
-Changes a rail's lockup parameters.
+Changes a rail's lockup parameters (`lockupPeriod` and `lockupFixed`).
 
-- **Parameters**:
-  - `railId`: Rail identifier
-  - `period`: New lockup period in epochs
-  - `lockupFixed`: New fixed lockup amount
-- **Requirements**:
-  - Caller must be the rail operator
-  - For terminated rails: cannot change period or increase fixed lockup
-  - For active rails: changes restricted if client's account isn't fully funded
-  - Operator must have sufficient allowances
+-   **When to use:** An operator calls this to adjust the client's funding guarantee. This is used to set an initial `lockupFixed` for an onboarding fee, increase the `lockupPeriod` for a longer-term commitment, or decrease lockups when a deal's terms change.
+
+**Lockup Calculation and State Changes**
+
+This function recalculates the rail's total lockup requirement based on the new `period` and `lockupFixed` values. The change in the rail's individual lockup is then applied to the client's total account lockup (`Account.lockupCurrent`).
+
+-   **State Impact:** It modifies both the `Rail` struct (updating `lockupPeriod` and `lockupFixed`) and the client's `Account` struct (updating `lockupCurrent`).
+-   **Safety Checks:** The entire operation is wrapped in a modifier that settles the client's account lockup before and after the change. This ensures that any increase in lockup is only permitted if the client has sufficient available funds to cover it, maintaining the contract's solvency invariants.
+
+-   **Parameters**:
```

Comment:
not sure these should be `-` dotpoints; maybe just lift them up as top-level things, otherwise they look like they are under "Lockup Caluclation and State Changes", same with `modifyRailPayment` below and `terminateRail` above